file_system/repository/file_repository.py

```diff
@@ -60,23 +59,62 @@ def filter(cls, queryset=None, path=None, path_in=[], path_regex=None, file_type
+            for k, queries in metadata.items():
+                if isinstance(queries,list):
+                    if len(queries) > 1:
+                        metadata_query_q = Q()
+                        for single_query in queries:
+                            single_query_dict = {'metadata__%s' % k: single_query}
+                            metadata_query_q |= Q(**single_query_dict)
+                        queryset = queryset.filter(metadata_query_q)
+                    else:
+                        v = queries[0]
+                        metadata_query_dict['metadata__%s' % k] = v
+                else:
+                    metadata_query_dict['metadata__%s' % k] = queries
         elif metadata_regex:
-            for k, v in metadata.items():
-                metadata_query_dict['metadata__%s__regex' % k] = v
+            for single_regex_line in metadata_regex:
+                regex_query_q = Q()
+                for k,v in single_regex_line:
+                    regex_query_dict = {'metadata__%s__regex' % k: v}
+                    regex_query_q |= Q(**regex_query_dict)
+                queryset = queryset.filter(regex_query_q)
         create_query_dict.update(metadata_query_dict)
         queryset = queryset.filter(
             Q(**create_query_dict) & Q(file__sample__redact=False)) if filter_redact else queryset.filter(
             **create_query_dict)
-
+        if exclude:
+            exc_dict = {}
+            for single_exclude_field in exclude:
+                exc_dict['metadata__%s__isnull' % single_exclude_field] = True
+            queryset = queryset.exclude(**exc_dict)
+        if distinct:
+            distinct_query = 'metadata__%s' % distinct
+            queryset = queryset.order_by(distinct_query).distinct(distinct_query)
+        if order_by and not values_metadata_list:
+            if distinct:
+                metadata_ids = queryset.values_list('id', flat=True)
+                queryset = FileRepository.all()
+                queryset = queryset.filter(id__in=metadata_ids)
+                queryset = queryset.order_by(order_by)
+            else:
+                queryset = queryset.order_by(order_by)
         if values_metadata:
             ret_str = 'metadata__%s' % values_metadata
             return queryset.values_list(ret_str, flat=True).order_by(ret_str).distinct(ret_str)
         if values_metadata_list:
             values_metadata_query_list = ['metadata__%s' % single_metadata for single_metadata in values_metadata_list]
             values_metadata_query_set = set(values_metadata_query_list)
             sorted_metadata_query_list = sorted(values_metadata_query_set)
-            return queryset.values_list(*sorted_metadata_query_list).order_by(sorted_metadata_query_list[0]).distinct()
-
+            metadata_ids = queryset.values_list('id', flat=True)
+            queryset = FileRepository.all()
+            queryset = queryset.filter(id__in=metadata_ids)
```

Comment:
This is primarily because of the distinct function. It forces a particular ordering and its not possible to change that ordering unless you create a new queryset.